### PR TITLE
Add 0x check for Address, Bytes, and Bytes32 in GraphQl requests

### LINKED ISSUE #811: GraphQL endpoint should strictly require 0x prefix on bytes
Per the GraphQL spec all `Bytes32`, `Bytes`, and `Address` values must be `0x` prefixed.  Current GraphQL parsing code accepts 0x prefixes but does not require them and silently adds them.

There should be at least 6 test cases in `ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql` to verify that each of the above field types produce an error response when provided with a non-prefixed value (3 cases) and each passes with a prefixed value (3 cases).

### PR DESCRIPTION
**Pull request by MakMuftic**
## Status
**READY**

## PR Description
Produce error response if any of `Address`, `Bytes` or `Bytes32` values
inside GraphQl requests are not prefixed with `0x` as defined by the spec.

Fix old tests that used values that are not prefixed with `0x` and add
additional tests to confirm `Address`, `Bytes` and `Bytes32` values will be
rejected if not prefixed with 0x.

Signed-off-by: Mak Muftic <mak@nodefactory.io>

## Fixed Issue(s)
fixes #811

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/internal/Scalars.java [hunk 1]
```diff
@@ -55,8 +55,13 @@ public Address parseLiteral(final Object input) throws CoercingParseLiteralExcep
           if (!(input instanceof StringValue)) {
             throw new CoercingParseLiteralException("Value is not any Address : '" + input + "'");
           }
+          String inputValue = ((StringValue) input).getValue();
+          if (!inputValue.startsWith("0x")) {
+            throw new CoercingParseLiteralException(
+                "Address value '" + inputValue + "' is not prefixed with 0x");
+          }
```
Comment (RatanRSur) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/internal/Scalars.java:62]: Just for the sake of keeping things logically co-located, I'd like to see this functionality in `Quantity.java` which has a lot of utility methods related to this. Maybe a `static` method like `Quantity.isValid(String string)`?

Comment (MakMuftic) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/internal/Scalars.java]: @RatanRSur thanks for the suggestion, I moved this check to `Qunatitiy.java` but as I read jdoc of class (_...Quantity fields are represented as minimal length hex strings with no zero-padding..._) I think this `Quantity.isValid(String string)` function should do additional checks

- is provided `string` valid hex string (_just check that characters are valid_)
- that provided hex string is not padded with `0`

or I could rename function to something that reflects what exactly is tested.

What do you think ?

**Comment by timbeiko** [comment]
@MakMuftic everything is ready to merge, but you'll need to merge in changes from master 😄

**Comment by MakMuftic** [comment]
@timbeiko now everything should be ready :smile:

**Pull request merged**